### PR TITLE
Refactor transformations

### PR DESCRIPTION
**Pull request by everzet**
This PR extracts individual transformations from `RepositoryArgumentTransformer` into individual classes under `Behat\Behat\Transformation\Transformation` namespace. This should make adding new transformations and fixing old ones a breeze.

This PR is basis for upcoming transformation improvements in 3.2.

### CODE REVIEW

File: src/Behat/Behat/Transformation/RegexGenerator.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * @author Konstantin Kudryashov <ever.zet@gmail.com>
+ */
+interface RegexGenerator
+{
+    /**
+     * Generates regular expression using provided parameters.
+     *
+     * @param string $suiteName
+     * @param string $pattern
+     * @param string$language
```
Comment (pamil) [inline src/Behat/Behat/Transformation/RegexGenerator.php]: Missing space.

File: src/Behat/Behat/Transformation/Transformation/RuntimeTransformation.php [hunk 2]
```diff
@@ -8,14 +8,16 @@
...
-namespace Behat\Behat\Transformation\Call;
+namespace Behat\Behat\Transformation\Transformation;
 
 use Behat\Behat\Transformation\Transformation;
 use Behat\Testwork\Call\RuntimeCallee;
 
 /**
  * Transformation that is created and executed in the runtime.
  *
+ * @deprecated Will be removed in 4.0. Use specific transformations instead
```
Comment (stof) [inline src/Behat/Behat/Transformation/Transformation/RuntimeTransformation.php]: I suggest using the ``@trigger_error('...', E_USER_DEPRECATED)`` convention used by Symfony (which reminds me that someone wrote a Behat extension to collect deprecations, which I said I would review and I forgot to do it)

Comment (stof) [inline src/Behat/Behat/Transformation/Transformation/RuntimeTransformation.php]: btw, if you rename the class, deprecating it is useless: it is already a BC break. you would need to keep it in the ``Call`` namespace

**Comment by everzet** [comment]
I'll move `RuntimeTransformation` back. I don't mind with triggering deprecation errors yet, it's not as current of occurrence as in Symfony ATM.

**Pull request merged**